src/utils.jl

```diff
@@ -38,18 +38,28 @@ function atanpos(y, x)
 end
 
 """
-    fixlon(lon)
+    checklat(lat)
+Checks if the latitude is in the range `[-90°,90°]` and returns it if yes or throws an error otherwise.
 
-Fix the longitude to be in the range `[-180°,180°]`.
+The check has a tolerance of `1e-6°` around +90° and -90°, so if `90° < abs(lat) < 90.000001°` the function returns either +90° or -90° depending on the sign of the input.
 """
-fixlon(lon) = ifelse(-180° ≤ lon ≤ 180°, lon, (lon % 360° + 540°) % 360° - 180°)
-
+function checklat(lat)
+  abslat = abs(lat)
+  T = typeof(abslat)
+  if abslat > 90° + 1e-6°
+    throw(ArgumentError("the latitude must be in the range [-90°,90°], while $lat was provided"))
+  end
+  ifelse(abslat > 90°, sign(lat) * T(90), lat)
+end
 """
-    islon180(lon)
+    fixlon(lon)
 
-Checks if the longitude is `180°` or `-180°`.
+Fix the longitude to be in the range `(-180°,180°]`.
 """
-islon180(lon) = abs(lon) == 180°
+function fixlon(lon) 
+  lon = rem(lon, 360°, RoundNearest)
+  ifelse(lon == -180°, 180°, lon)
```

Comment:
I believe this implementation has a type instability. Here is my suggestion to fix it:
```suggestion
  ifelse(lon == -180°, oftype(lon, 180°), lon)
```

Reply:
@eliascarv you are right. I weirdly noticed that implementing this change slightly degrades performance on 1.11 (goes back to ~210μs in my PC vs the ~170μs without oftype), I do not really understand why this since there is indeed some yellow output from `@code_warntype` without `oftype`. I applied the change anyhow to be more correct